### PR TITLE
CE-1623 Adding tracking for special insights index page

### PR DESCRIPTION
**Pull request by kamilkoterba**
https://wikia-inc.atlassian.net/browse/CE-1623

Added tracking for index page of Special:Insights.
Reorganized structure of loading tracking scripts to load only on pages where it's used.

@Wikia/community-engineering

### CODE REVIEW

File: extensions/wikia/Insights/InsightsHooks.class.php [hunk 1]
```diff
@@ -13,12 +13,13 @@ public static function onBeforePageDisplay( OutputPage &$out, Skin &$skin ) {
...
 		}
 
-		// Load scripts for pages in insights loop and Special:Insights
-		if ( InsightsHelper::isInsightPage( $subpage ) || F::app()->wg->title->isSpecial( 'Insights' ) ) {
-			$out->addScriptFile( '/extensions/wikia/Insights/scripts/Insights.run.js' );
-			$out->addScriptFile( '/extensions/wikia/Insights/scripts/LoopTracking.js' );
+		// Load scripts for Special:Insights
+		if ( F::app()->wg->title->isSpecial( 'Insights' ) ) {
+			$out->addScriptFile( '/extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js' );
+			$out->addScriptFile( '/extensions/wikia/Insights/scripts/InsightsListTracking.js' );
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/InsightsHooks.class.php]: I think we can add these scripts to Insights Controller

File: extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+				action: tracker.ACTIONS.CLICK_LINK_TEXT,
+				label: $(this).parent().data('type')
+			};
+			tracker.track(trackingParams);
+		}
+
+		function init() {
+			/* Setup click events on all links within special insights container section */
+			var $insightsContainer = $('.insights-container-landing');
+			$insightsContainer.find('a').each(function(){
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js]: Can we use ```on``` function and bind click event once? I will also use 'mousedown' event

File: extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+				$(this).click(onLinkClick);
+			});
+		}
+
+		init();
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js]: Shouldn't we run this function on dom ready?

File: extensions/wikia/Insights/scripts/InsightsListTracking.js [hunk 4]
```diff
@@ -27,8 +23,6 @@ define(
 			});
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsListTracking.js]: Can we use on function and bind click event once? I will also use 'mousedown' event

File: extensions/wikia/Insights/scripts/InsightsListTracking.js [hunk 5]
```diff
@@ -27,8 +23,6 @@ define(
...
 		}
 
-		return {
-			init: init
-		};
+		init();
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsListTracking.js]: Shouldn't we run this function on dom ready?

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+ * checks for item_status url param to log action status (fixed/notfixed)
+ */
+define('ext.wikia.Insights.LoopNotificationTracking',
+	['jquery', 'wikia.tracker', 'wikia.querystring'],
+	function ($, tracker, Querystring) {
+		'use strict';
+
+		var qs = new Querystring(),
+			insightType = qs.getVal('insights', null),
+			itemStatus = qs.getVal('item_status', null);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: I know part with VE is not finished now but we need to consider how to pass this param when VE is enabled

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+		/* If there's no itemStatus use action parameter */
+		if (itemStatus === null) {
+			itemStatus = qs.getVal('action', null);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: Why you get 'action' param here?

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+				};
+				tracker.track(trackingParams);
+			}
+		}
+
+		function init(event, bannerNotification) {
+			/* Setup click events within BannerNotification */
+			bannerNotification.$element.find('#InsightsBackToListButton').click('back-to-list', onClickTrack);
+			bannerNotification.$element.find('#InsightsNextPageButton').click('next-page', onClickTrack);
+			bannerNotification.$element.find('button.close').click('dismiss', onClickTrack);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: I will consider using 'mousedown' event to send tracking request earlier

File: extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+				label: insightType+'-'+itemStatus+'-'+linkType.data
+			};
+			tracker.track(trackingParams);
+		}
+
+		/**
+		 * Add log for success banner impression if status is fixed
+		 */
+		function successTrack() {
+			if(itemStatus==='fixed') {
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsLoopNotificationTracking.js]: Shouldn't we track both statuses?

File: extensions/wikia/Insights/scripts/LoopNotification.js [hunk 10]
```diff
@@ -21,9 +28,12 @@ require(['jquery', 'BannerNotification', 'wikia.querystring', 'wikia.window'],
 	showNotification = function(html) {
 		if (html) {
 			var msgType = getNotificationType(),
-				$parent = getParent();
+				$parent = getParent(),
+				bn;
 
-			new BannerNotification(html, msgType, $parent).show();
+			bn = new BannerNotification(html, msgType, $parent);
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/LoopNotification.js]: Can we create this object when we declare bn var?

File: extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js [hunk 11]
```diff
@@ -0,0 +1,26 @@
+require(['wikia.tracker'],
+	function (tracker) {
+		'use strict';
+
+		function onLinkClick() {
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js]: What do you think about using ```buildTrackingFunction``` from wikia tracker and prepare params outside ```onLinkClick``` function and pass only label?

File: extensions/wikia/Insights/scripts/InsightsListTracking.js [hunk 12]
```diff
@@ -1,18 +1,14 @@
...
-	[
-		'wikia.tracker'
-	],
+require(['wikia.tracker'],
 	function (tracker) {
 		'use strict';
 
 		var insightType;
 
 		function onLinkClick() {
```
Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsListTracking.js]: What do you think about using ```buildTrackingFunction``` from wikia tracker and prepare params outside ```onLinkClick``` function and pass only label?

Comment (lukaszkonieczny) [inline extensions/wikia/Insights/scripts/InsightsListTracking.js]: Actually if we are adding both files (this and ```InsightsIndexPageTracking```) on Special Insights, maybe we can combine them if we hava the same functions in both

File: extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js (hunk 3 above)
Comment (kamilkoterba) [inline extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js]: true

File: extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js (hunk 2 above)
Comment (kamilkoterba) [inline extensions/wikia/Insights/scripts/InsightsIndexPageTracking.js]: ok

**Pull request merged**